dev/setup_localstack.py

```diff
@@ -0,0 +1,192 @@
+
+    def get_security_group_id(self):
+        print("Getting security group ID")
+        response = self.ec2_client.describe_security_groups()
+        security_groups = response.get("SecurityGroups", [""])
+        group_id = security_groups[0].get("GroupId", "")
+        print(f"Got group ID {group_id}")
+        return group_id
+
+    def create_subnet(self):
+        print("Creating VPC and subnet")
+        response = self.ec2_client.create_vpc(
+            CidrBlock="10.0.0.0/16"
+        )
+        vpc_info = response.get("Vpc", {})
+        vpc_id = vpc_info.get("VpcId", {})
+        self.ec2_client.create_subnet(
+            CidrBlock="10.0.2.0/24",
+            VpcId=vpc_id
+        )
+        print("Created VPC and subnet")
+
+    def get_subnet_id(self):
+        print("Getting subnet ID")
+        response = self.ec2_client.describe_subnets()
+        subnets = response.get("Subnets", [{}])
+        subnet_id = subnets[0].get("SubnetId", "")
+        print(f"Got subnet ID {subnet_id}")
+        return subnet_id
+
+    def create_compute_environment(self, security_group_id, batch_service_role_arn, subnet_id):
+        print(f"Creating compute environment for security group {security_group_id}")
+        BATCH_COMPUTE_RESOURCES["securityGroupIds"].append(security_group_id)
+        BATCH_COMPUTE_RESOURCES["subnets"].append(subnet_id)
+        self.batch_client.create_compute_environment(
+            computeEnvironmentName=BATCH_COMPUTE_ENVIRONMENT_NAME,
+            type=BATCH_COMPUTE_ENVIRONMENT_TYPE,
+            computeResources=BATCH_COMPUTE_RESOURCES,
+            serviceRole=batch_service_role_arn
+        )
+        print("Created security group")
+
+    def create_batch_job_queue(self, queue_name):
+        print(f"Creating job queue {queue_name}")
+        self.batch_client.create_job_queue(
+            jobQueueName=queue_name,
+            priority=1,
+            state="ENABLED",
+            computeEnvironmentOrder=BATCH_COMPUTE_ENVIRONMENT_ORDER
+        )
+        print("Created job queue")
+
+    def create_s3_bucket(self, bucket_name):
+        print(f"Creating bucket {bucket_name}")
+        self.s3_client.create_bucket(
+            Bucket=bucket_name
+        )
+        print("Created bucket")
+
+    def setup_ses(self, email_address):
+        print(f"Verifying email address {email_address}")
+        self.ses_client.verify_email_identity(
+            EmailAddress=email_address
+        )
+        print("Verified email address")
+
+
+def wait_for_localstack():
+    healthcheck_url = "".join([LOCALSTACK_URL, "/health"])
+    counter = 0
+    while True and counter < 15:
+        try:
+            response = requests.get(healthcheck_url)
+            if BATCH_READY in response.text:
+                return
+        except requests.exceptions.ConnectionError:
+            pass
+        counter += 1
+        print("Waiting for localstack")
+        sleep(5)
+    raise Exception("Localstack not available")
```

Comment:
When I try running `dev/run_full_stack.sh` locally I hit this exception, is there any setup I need to do? I've set `LOCALSTACK_API_KEY` in `dev/.env`.

Reply:
I suspect a timing issue. I increased the number of attempts to 20.

Reply:
Can't reproduce this, but I got this error message:

```Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /tmp/localstack```

Creating /tmp/localstack fixed this

Reply:
> Creating /tmp/localstack fixed this

Should that ideally go in the host's `/tmp` or the docker image's `/tmp`? Either way, it needs to be scripted.

Reply:
`/tmp` exists in the docker image, but I'm not using it for anything--the volume declared in the compose file was a default--so I removed it.

Reply:
I still see that the timeout exception is raised before the localstack has launched:

```
setup-localstack_1  | Waiting for localstack
setup-localstack_1  | Traceback (most recent call last):
setup-localstack_1  |   File "./setup_localstack.py", line 177, in <module>
setup-localstack_1  |     wait_for_localstack()
setup-localstack_1  |   File "./setup_localstack.py", line 172, in wait_for_localstack
setup-localstack_1  |     raise Exception("Localstack not available")
setup-localstack_1  | Exception: Localstack not available
setup-localstack_1 exited with code 1
localstack_1        | Waiting for all LocalStack services to be ready
localstack_1        | 2021-07-16T08:25:19:INFO:botocore.credentials: Found credentials in environment variables.
localstack_1        | Waiting for all LocalStack services to be ready
localstack_1        | Ready.
localstack_1        | 2021-07-16 08:25:24,808:API: 127.0.0.1 - - [16/Jul/2021 08:25:24] "GET / HTTP/1.1" 200 -
localstack_1        | 2021-07-16T08:25:24:INFO:localstack.utils.analytics.profiler: Execution of "start_api_services" took 12558.80ms
```

Could the setup image use [`wait-for-it.sh`](https://github.com/vishnubob/wait-for-it/blob/master/wait-for-it.sh) and run when the localstack service is finally accepting connections?